File: EcoQuestAPI/src/main/java/com/proyecto/spring/dto/MensajeDTO.java
package com.proyecto.spring.dto;

import com.proyecto.spring.modelos.Mensaje;

public class MensajeDTO {

    public Long id;
    public String texto;
    public String fechaHora;
    public Long usuarioId;
    public String usuarioNombre;
    public String usuarioNombreUsuario;
    public String usuarioImagen;
    public Long comunidadId;

    public MensajeDTO() {}

    public MensajeDTO(Mensaje m) {
        this.id = m.getId();
        this.texto = m.getTexto();
        this.fechaHora = m.getFechaHora() != null ? m.getFechaHora().toString() : null;
        this.usuarioId = m.getUsuario() != null ? m.getUsuario().getId() : null;
        this.usuarioNombre = m.getUsuario() != null ? m.getUsuario().getNombre() + " " + m.getUsuario().getApellido() : null;
        this.usuarioNombreUsuario = m.getUsuario() != null ? m.getUsuario().getNombreUsuario() : "Usuario";
        this.usuarioImagen = m.getUsuario() != null ? m.getUsuario().getImagen() : null;
        this.comunidadId = m.getComunidad() != null ? m.getComunidad().getId() : null;
    }
}

File: EcoQuestMobile/app/src/main/java/com/ecoquest/app/data/remote/dto/MensajeDto.kt
package com.ecoquest.app.data.remote.dto

import com.google.gson.annotations.SerializedName

data class MensajeDto(
    @SerializedName("id")                val id: Long = 0,
    @SerializedName("texto")             val texto: String = "",
    @SerializedName("fechaHora")         val fechaHora: String = "",
    @SerializedName("usuarioId")         val usuarioId: Long = 0,
    @SerializedName("usuarioNombre")     val usuarioNombre: String = "",
    @SerializedName("usuarioNombreUsuario") val usuarioNombreUsuario: String = "",
    @SerializedName("usuarioImagen")     val usuarioImagen: String? = null,
    @SerializedName("comunidadId")       val comunidadId: Long = 0
)
